File: math-problems/reverse.py
def reverse(x: int) -> int:
    if x >= pow(2,31)-1 or x < -pow(2,31):
        return 0

    y = str(x)
    signed = False
    if y[0] == '-':
        y = y[1:]
        signed = True
    z = ""
    if signed:
        z = '-'
    for i in range(len(y)-1,-1,-1):
        z+=y[i]
    return int(z)

print(reverse(-123))

File: math-problems/reverse.rs
fn main() {
    let result = reverse(-123);
    println!("{0}", result);
}

fn reverse(x: i32) -> i32 {
    if x >= 2147483647 || x < -2147483648 {
        return 0;
    }
    println!("{}", x);

    let y = x.to_string();
    let mut signed = false;
    let mut z = String::new();
    if y.chars().nth(0).unwrap() == '-' {
        z = y[1..].to_string();
        signed = true;
    }

    for i in (0..y.len()).rev() {
        z.push(y.chars().nth(i).unwrap());
    }

    if signed {
        return -z.parse::<i32>().unwrap();

    } else {
        return z.parse::<i32>().unwrap();
    }
    
}
